src/shell/shell.c

```diff
@@ -433,14 +433,15 @@ static json_t *flux_shell_get_info_object (flux_shell_t *shell)
         return o;
 
     if (!(o = json_pack_ex (&err, 0,
-                            "{ s:I s:i s:i s:i s:s s:O s:O s:{ s:i s:b }}",
+                            "{ s:I s:i s:i s:i s:s s:O s:O s:s s:{ s:i s:b }}",
                             "jobid", shell->info->jobid,
                             "rank",  shell->info->shell_rank,
                             "size",  shell->info->shell_size,
                             "ntasks", shell->info->total_ntasks,
                             "service", shell_svc_name (shell->svc),
                             "jobspec", shell->info->jobspec->jobspec,
                             "R", shell->info->R,
+                            "hwloc", shell->info->hwloc_xml,
```

Comment:
This is likely a minor consideration compared with the big performance gains attained here, but since the shell info object is accessed as a string, it would avoid unnecessary json encoding/decoding to store the hwloc xml under a separate aux key and add a specialized accessor that just passes the string through.

Since the hwloc xml can be large, and any access to the shell info object from a plugin would trigger its decode, it might be worth it.

Reply:
Oops, sorry I misread the code - the `json_dumps()` only happens if a plugin calls `flux_shell_get_info()` to access the object for parsing with another JSON library.  The unpack functions access the raw json object directly.

If the XML string is merely attached to a json object that is never serialized, then I'm not sure there is any downside at all to doing it this way,  other than jansson checking that the string is utf8 and scanning for the null terminator.  Even that could be avoided by packing with `s#` or `s%` if the XML content encoding says utf8.  Meh, premature optimization.